checkpoint/recovery.go

```diff
@@ -315,25 +319,22 @@ func collectOwnAtxDeps(
 	}
 	var ref types.ATXID
 	var own bool
-	if atxid == types.EmptyATXID {
-		if m, _ := initialization.LoadMetadata(cfg.PostDataDir); m != nil {
-			ref = types.ATXID(types.BytesToHash(m.CommitmentAtxId))
-			logger.With().Debug("found commitment atx from metadata",
-				log.Stringer("commitment atx", ref),
-			)
-		}
-	} else {
+	if atxid != types.EmptyATXID {
 		ref = atxid
-		logger.With().Debug("found own atx",
-			log.Stringer("own atx", ref),
-		)
+		logger.With().Debug("found own atx", log.Stringer("own atx", ref))
 		own = true
 	}
 
 	// check for if miner is building any atx
-	nipostCh, _ := activation.LoadNipostChallenge(cfg.PostDataDir)
-	if ref == types.EmptyATXID && nipostCh == nil {
-		return nil, nil, nil
+	nipostCh, _ := nipost.Challenge(localDb, cfg.NodeID)
+	if ref == types.EmptyATXID {
+		if nipostCh == nil {
+			return nil, nil, nil
+		}
+		if nipostCh.CommitmentATX != nil {
+			ref = *nipostCh.CommitmentATX
+			// TODO(mafa): if nil query post service for commitment atx (in case it is currently initializing)
```

Comment:
For a local post service this works, but we cannot guarantee that any remote post service is online while we are checkpointing.